test/core/miner/precommit_batcher_test.cpp

```diff
@@ -0,0 +1,210 @@
+        }
+
+        return ERROR_TEXT("ERROR");
+      };
+
+      EXPECT_CALL(*sch_, now()).WillOnce(testing::Return(current_time_));
+      auto result = PreCommitBatcherImpl::makeBatcher(
+          toTicks(std::chrono::seconds(60)), api_, sch_, miner_address_);
+      batcher_ = result.value();
+      ASSERT_FALSE(result.has_error());
+    }
+
+    std::shared_ptr<FullNodeApi> api_;
+    std::shared_ptr<SchedulerMock> sch_;
+    std::shared_ptr<PreCommitBatcherImpl> batcher_;
+    std::shared_ptr<Tipset> tipset_;
+    Address miner_address_;
+    uint64_t miner_id_;
+    RegisteredSealProof seal_proof_type_;
+    Ticks current_time_;
+    TokenAmount mutual_deposit_;
+    bool is_called_;
+  };
+
+  TEST_F(PreCommitBatcherTest, BatcherWrite) {
+    SectorInfo si = SectorInfo();
+    api::SectorPreCommitInfo precInf;
+    TokenAmount deposit = 10;
+    EXPECT_OUTCOME_TRUE_1(batcher_->addPreCommit(si, deposit, precInf, [](outcome::result<CID>){}));
+  };
+
+  /**
+   * CallbackSend have 4 unic precommits sent in pairs
+   * a PrecommitBatcher should send the first two and after the rescheduling
+   * next pair the result should be 2 messages in message pool with pair of
+   * precommits in each
+   */
+  TEST_F(PreCommitBatcherTest, CallbackSend) {
+    mutual_deposit_ = 0;
+    is_called_ = false;
+
+    SectorInfo si = SectorInfo();
+    api::SectorPreCommitInfo precInf;
+    TokenAmount deposit = 10;
+
+    precInf.sealed_cid = "010001020005"_cid;
+    si.sector_number = 2;
+
+    EXPECT_OUTCOME_TRUE_1(batcher_->addPreCommit(si, deposit, precInf, [](outcome::result<CID>){}));
+    mutual_deposit_ += 10;
+
+    precInf.sealed_cid = "010001020006"_cid;
+    si.sector_number = 3;
+
+    EXPECT_OUTCOME_TRUE_1(batcher_->addPreCommit(si, deposit, precInf, [](outcome::result<CID>){}));
+    mutual_deposit_ += 10;
+
+    EXPECT_CALL(*sch_, now())
+        .WillOnce(
+            testing::Return(current_time_ + toTicks(std::chrono::seconds(61))))
+        .WillOnce(
+            testing::Return(current_time_ + toTicks(std::chrono::seconds(123))))
+        .WillRepeatedly(testing::Return(current_time_
+                                        + toTicks(std::chrono::seconds(300))));
+    sch_->next_clock();
+    ASSERT_TRUE(is_called_);
+
+    is_called_ = false;
+    mutual_deposit_ = 0;
+
+    precInf.sealed_cid = "010001020008"_cid;
+    si.sector_number = 6;
+
+    EXPECT_OUTCOME_TRUE_1(batcher_->addPreCommit(si, deposit, precInf, [](outcome::result<CID>){}));
+    mutual_deposit_ += 10;
+    sch_->next_clock();
+    ASSERT_TRUE(is_called_);
+    is_called_ = false;
+  }
+
+  /**
+   * ShortDistanceSending have 3 Precommits
+   * Test should send first two after 30 sec instead of 60 and one more
+   *immediately after the addition The result should be 2 new messages in
+   *message pool 1st: 2 precommits, have been sent after 30  sec, and 2nd: one
+   *precommmit, have been sent after the first one immediately
```

Comment:
```suggestion
   * ShortDistanceSending have 3 Precommits
   * @given First two are sent after 30 sec instead of 60 and one more
   * immediately after the addition.
   * @then The result should be 2 new messages in message pool 1st: 
   * 2 precommits, have been sent after 30  sec, and 2nd: one
   * precommmit, have been sent after the first one immediately
```

Reply:
fixed